File: src/main/java/fgo/cards/fgo/Indomitable.java
package fgo.cards.fgo;

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.monsters.AbstractMonster;

import fgo.cards.FGOCard;
import fgo.powers.IndomitablePower;
import fgo.powers.NonStackableGutsPower;

public class Indomitable extends FGOCard {
    public static final String ID = makeID(Indomitable.class.getSimpleName());

    public Indomitable() {
        super(ID, 1, CardType.POWER, CardTarget.SELF, CardRarity.RARE);
        setMagic(5, 5);
    }

    @Override
    public void use(AbstractPlayer p, AbstractMonster m) {
        addToBot(new ApplyPowerAction(p, p, new NonStackableGutsPower(p, magicNumber, 1)));
        addToBot(new ApplyPowerAction(p, p, new IndomitablePower(p, 2)));
    }
}

File: Scripts/Cards/Indomitable.cs
using Fgo.Scripts.Powers;
using MegaCrit.Sts2.Core.Commands;
using MegaCrit.Sts2.Core.Entities.Cards;
using MegaCrit.Sts2.Core.GameActions.Multiplayer;
using MegaCrit.Sts2.Core.HoverTips;
using MegaCrit.Sts2.Core.Localization.DynamicVars;
using STS2RitsuLib.Cards.DynamicVars;

namespace Fgo.Scripts.Cards;

public class Indomitable() : FgoCardModel(1, CardType.Power,
    CardRarity.Rare, TargetType.Self)
{
    protected override IEnumerable<IHoverTip> AdditionalHoverTips =>
    [
        HoverTipFactory.FromPower<NonStackableGutsPower>(),
        HoverTipFactory.FromPower<IndomitablePower>()
    ];

    protected override IEnumerable<DynamicVar> CanonicalVars =>
    [
        ModCardVars.Power<NonStackableGutsPower>(5),
        ModCardVars.Power<IndomitablePower>(2)
    ];

    protected override void OnUpgrade()
    {
        DynamicVars[nameof(NonStackableGutsPower)].UpgradeValueBy(10);
    }

    protected override async Task OnPlay(
        PlayerChoiceContext choiceContext,
        CardPlay play)
    {
        await PowerCmd.Apply<NonStackableGutsPower>(choiceContext, Owner.Creature,
            DynamicVars[nameof(NonStackableGutsPower)].BaseValue, Owner.Creature, this);
        await PowerCmd.Apply<IndomitablePower>(choiceContext, Owner.Creature,
            DynamicVars[nameof(IndomitablePower)].BaseValue, Owner.Creature, this);
    }
}